analyzers/src/SonarAnalyzer.Common/Rules/Utilities/TokenTypeAnalyzerBase.cs

```diff
@@ -34,36 +34,62 @@ public abstract class TokenTypeAnalyzerBase<TSyntaxKind> : UtilityAnalyzerBase<T
 
         protected TokenTypeAnalyzerBase() : base(DiagnosticId, Title) { }
 
-        protected abstract TokenClassifierBase GetTokenClassifier(SyntaxToken token, SemanticModel semanticModel, bool skipIdentifierTokens);
+        protected abstract TokenClassifierBase GetTokenClassifier(SemanticModel semanticModel, bool skipIdentifierTokens);
+        protected abstract TriviaClassifierBase GetTriviaClassifier();
 
         protected sealed override TokenTypeInfo CreateMessage(SyntaxTree syntaxTree, SemanticModel semanticModel)
         {
             var tokens = syntaxTree.GetRoot().DescendantTokens();
             var identifierTokenKind = Language.SyntaxKind.IdentifierToken;  // Performance optimization
-            var skipIdentifierTokens = tokens.Count(token => Language.Syntax.IsKind(token, identifierTokenKind)) > IdentifierTokenCountThreshold;
+            var skipIdentifierTokens = tokens
+                .Where(token => Language.Syntax.IsKind(token, identifierTokenKind))
+                .Take(IdentifierTokenCountThreshold + 1)
+                .Count() > IdentifierTokenCountThreshold;
 
+            var tokenClassifier = GetTokenClassifier(semanticModel, skipIdentifierTokens);
+            var triviaClassifier = GetTriviaClassifier();
             var spans = new List<TokenTypeInfo.Types.TokenInfo>();
             // The second iteration of the tokens is intended since there is no processing done and we want to avoid copying all the tokens to a second collection.
             foreach (var token in tokens)
             {
-                spans.AddRange(GetTokenClassifier(token, semanticModel, skipIdentifierTokens).Spans);
+                if (token.HasLeadingTrivia)
+                {
+                    IterateTrivia(token.LeadingTrivia);
+                }
+                if (tokenClassifier.ClassifyToken(token) is { } tokenClassification)
+                {
+                    spans.Add(tokenClassification);
+                }
+                if (token.HasTrailingTrivia)
+                {
+                    IterateTrivia(token.TrailingTrivia);
+                }
```

Comment:
We are on the hot path here. I changed the implementation by making IterateTrivia static and passing `spans` to it.

I'd like to avoid as many allocations and work as possible here and therefore, keep the procedure approach (e.g., the HasTrivia check to avoid the request to the Trivia property).